src/components/SearchModal/CurrencySearch.tsx

```diff
@@ -90,33 +91,52 @@ export function CurrencySearch({
     return Object.values(defaultTokens).filter(getTokenFilter(debouncedQuery))
   }, [defaultTokens, debouncedQuery])
 
-  const [balances, balancesAreLoading] = useAllTokenBalances()
+  const { data, loading: balancesAreLoading } = useCachedPortfolioBalancesQuery({ account })
```

Comment:
how will this degrade if the backend is unresponsive?

Reply:
it will just show 0 for all balances and use the default list order. before this, if the multicall balance checks failed we would show infinite loading  spinners.